pkg/demoinfocs/game_events.go

```diff
@@ -119,7 +119,17 @@ func (geh gameEventHandler) gameState() *gameState {
 }
 
 func (geh gameEventHandler) playerByUserID(userID int) *common.Player {
-	return geh.gameState().playersByUserID[userID]
+	player := geh.gameState().playersByUserID[userID]
+	if player != nil || !geh.parser.isSource2() {
```

Comment:
I don't change it for Source 1 demos because in case of damages/kills done by the "world" it would return the GOTV entity (id `0`) instead of nil.

In Source 2 the CSTV entity ID is not always 0 and may have `SourceTV` or `DemoAutoRecorder` as a name.